Add Orca-Math

mmhamdy opened the pull request:
Hi,
This PR adds the Orca-Math Dataset (with the downloader, preparer, and mapper). All tests passed successfully.

shayne-longpre commented on data_summaries/Orca-Math.json:
```diff
@@ -0,0 +1,66 @@
+        ],
+        "Text Sources": [
+            "arithmetic word problems",
+            "Grade School Math"
```
Is this a textbook?

shayne-longpre replied:
We need to do a bit more digging here. It looks like the questions are sourced from several datasets. For each of these, can we trace what sources/models the data was collected from (websites, certain textbooks, or something else)?

"We start by collecting sample math word problems from existing open-source
datasets, namely NumGLUE [26], AddSub [13], ALGES [17], ASDiv [24], DRAW [35],
GSM8k [7], MATHQA [2], MultiArith [32], SingeOP [33], SingleEQ [16], and SVAMP [30].
We collect a total of 36, 217 problems."

mmhamdy replied:
On it!

shayne-longpre commented on data_summaries/Orca-Math.json:
```diff
@@ -0,0 +1,66 @@
+        ],
+        "Licenses": [
+            {
+                "License": "MIT License",
```
We will need to add licenses here for the datasets it was derived from.

shayne-longpre reviewed (approved):
We will have to expand the licenses and sources list, but otherwise looks good!

mmhamdy commented:
I've traced the source datasets and added the licenses I could find. Unfortunately, most datasets that were released before 2017 don't even have a dataset card and just provide a simple link without any context.

- Those are the datasets I was able to find licenses for: `NumGLUE`, `ASDiv`, `GSM8K`, `MATHQA`, `Lila`
- Also, `Lila` is derived from most of the above-mentioned datasets.
- I've expanded the text sources with the sources that were used in the source datasets.

I've also added a new license class for `NumGLUE` guided by the information provided [here](https://opendatacommons.org/licenses/by/1-0/)
